runbotics-orchestrator/src/main/java/com/runbotics/service/impl/ProcessServiceImpl.java

```diff
@@ -76,7 +78,21 @@ public ProcessDTO save(ProcessDTO processDTO) {
             BotSystem any = new BotSystem(BotSystem.BotSystemName.ANY.value());
             process.setSystem(any);
         }
+
+        if (process.getTags() != null) {
+            if (process.getTags().size() > 15) {
+                throw new BadRequestAlertException("Tag limit of 15 exceeded", ENTITY_NAME, "tooManyTags");
+            }
+
+            Set<Tag> tags = tagService.processTags(processDTO.getTags());
+            process.setTags(tags);
+        }
+
         process = processRepository.save(process);
+        if (processDTO.getId() != null) {
+            List<Long> remainingTags = tagService.checkTagsToDelete(processDTO);
+            tagService.deleteUnusedTags(remainingTags);
+        }
```

Comment:
What does it check? Does it matter here?

imo it should occur regardless of edition or creation + those 2 operations you can do with just 1

Reply:
bump